test/OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.test.cpp

```diff
@@ -0,0 +1,367 @@
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/CartesianVelocity.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.hpp>
+
+#include <Global.test.hpp>
+
+using ostk::core::types::Shared;
+using ostk::core::ctnr::Array;
+
+using ostk::math::obj::VectorXd;
+
+using ostk::physics::coord::Frame;
+using ostk::physics::time::DateTime;
+using ostk::physics::time::Instant;
+using ostk::physics::time::Scale;
+
+using ostk::astro::trajectory::State;
+using ostk::astro::trajectory::StateBuilder;
+using ostk::astro::trajectory::state::CoordinatesBroker;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianPosition;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianVelocity;
+
+class OpenSpaceToolkit_Astrodynamics_Trajectory_StateBuilder : public ::testing::Test
+{
+   protected:
+    const Array<Shared<const CoordinatesSubset>> posVelSubsets = {
+        CartesianPosition::Default(), CartesianVelocity::Default()
+    };
+
+    const Array<Shared<const CoordinatesSubset>> posVelMassSubsets = {
+        CartesianPosition::Default(),
+        CartesianVelocity::Default(),
+        CoordinatesSubset::Mass(),
+    };
+
+    const Shared<const CoordinatesBroker> posVelBrokerSPtr =
+        std::make_shared<CoordinatesBroker>(CoordinatesBroker(posVelSubsets));
+
+    const Shared<const CoordinatesBroker> posVelMassBrokerSPtr =
+        std::make_shared<CoordinatesBroker>(CoordinatesBroker(posVelMassSubsets));
+};
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_StateBuilder, Constructor)
+{
+    {
+        EXPECT_NO_THROW(StateBuilder builder(Frame::GCRF(), posVelBrokerSPtr));
+    }
+
+    {
+        EXPECT_NO_THROW(StateBuilder builder(Frame::GCRF(), posVelSubsets););
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_StateBuilder, EqualToOperator)
+{
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+
+        EXPECT_TRUE(aStateBuilder == anotherStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelSubsets};
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+
+        EXPECT_TRUE(aStateBuilder == anotherStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+        const StateBuilder anotherStateBuilder = {Frame::ITRF(), posVelBrokerSPtr};
+
+        EXPECT_FALSE(aStateBuilder == anotherStateBuilder);
+    }
+
+    {
+        const StateBuilder aStateBuilder = {Frame::GCRF(), posVelBrokerSPtr};
+        const StateBuilder anotherStateBuilder = {Frame::GCRF(), posVelMassBrokerSPtr};
+
+        EXPECT_FALSE(aStateBuilder == anotherStateBuilder);
+        EXPECT_FALSE(anotherStateBuilder == aStateBuilder);
```

Comment:
leftover line?

Reply:
I the current comparison, it checks that each contain each other, so the reverse comparison is needed for full coverage